test/runtests.jl

```diff
@@ -28,554 +33,746 @@ mycount(p, c) = p + 1
 bigsum(p, c) = p + big(c)
 sumpoint(p::Point3D, x, y, z) = p + Point3D(x, y, z)
 
-dbfile = joinpath(dirname(pathof(SQLite)), "../test/Chinook_Sqlite.sqlite")
-dbfile2 = joinpath(tempdir(), "test.sqlite")
-cp(dbfile, dbfile2; force=true)
-chmod(dbfile2, 0o777)
+test_dbfile = joinpath(tempdir(), "test.sqlite")
 
-@testset "basics" begin
+"""
+    setup_clean_test_db()
+
+Copy test sqlite file to temp directory, path `test.sqlite`,
+overwrite file if necessary and set permissions.
+"""
+function setup_clean_test_db()
```

Comment:
So looking at the failing windows tests, here's what I think is going on: we used to only call, effectively, `setup_clean_test_db` once and use it in a number of tests. Now, we're calling `setup_clean_test_db` multiple times, but on windows, when we get to the `cp(dbfile, test_Dbfile; force = true)` line, it fails because it's an error to try and remove/delete a file that is actively being used. So because the previous testset `SQLite.DB` object still has a reference to the underlying file object, we get the error.

Couple ideas on how to fix:
  * just call `setup_clean_test_db` once like we were doing before, make it a runtests/test.jl global variable and re-use it in testsets
  * close the `SQLite.DB` object at the end of each testset; we might have to call `GC.gc()` as well to ensure everything is properly cleaned up/finalized and the underlying file object is "released"

Reply:
Great! Thanks for the feedback, only was able to test locally on a Mac. :) It highlights a way in which the test setup is sloppy, so I'll look into your second suggestion on how to fix it, but I definitely want to make each test set idempotent, so I think we want to setup and tear down per test set.

Reply:
I think what we needed was to turn the function into a do block. Seems to work on Mac, curious to see what the CI/CD has to say. :)